resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -590,6 +601,10 @@ protected EventLoop executor() {
         return (EventLoop) super.executor();
     }
 
+    private InetAddress resolveHostsFileEntry(String hostname) {
+        return hostsFileEntriesResolver != null ? hostsFileEntriesResolver.getAddress(hostname) : null;
```

Comment:
cache hostsFileEntriesResolver in this method so we not need to have 2 volatile access.

```java
HostsFileEntriesResolver hostsFileEntriesResolver = this. hostsFileEntriesResolver;
return hostsFileEntriesResolver != null ? hostsFileEntriesResolver.getAddress(hostname) : null;
```